sdk/communication/azure-communication-identity/src/main/java/com/azure/communication/identity/CommunicationIdentityAsyncClient.java

```diff
@@ -113,20 +115,57 @@ public Mono<Response<CommunicationUserIdentifier>> createUserWithResponse() {
      * Creates a new CommunicationUserIdentifier with token.
      *
      * @param scopes The list of scopes for the token.
+     * @param tokenExpiresAfter Custom validity period of the Communication Identity access token within [1,24]
+     * hours range. If not provided, the default value of 24 hours will be used.
      * @return The created communication user and token.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<CommunicationUserIdentifierAndToken>
-        createUserAndToken(Iterable<CommunicationTokenScope> scopes) {
+        createUserAndToken(Iterable<CommunicationTokenScope> scopes, Duration tokenExpiresAfter) {
         try {
             Objects.requireNonNull(scopes);
             final List<CommunicationTokenScope> scopesInput = StreamSupport.stream(scopes.spliterator(), false).collect(Collectors.toList());
-            return client.createAsync(new CommunicationIdentityCreateRequest().setCreateTokenWithScopes(scopesInput))
+
+            CommunicationIdentityCreateRequest communicationIdentityCreateRequest = new CommunicationIdentityCreateRequest();
+            communicationIdentityCreateRequest.setCreateTokenWithScopes(scopesInput);
+
+            int expiresInMinutes;
+
+            if (tokenExpiresAfter != null) {
```

Comment:
you can reuse `getTokenExpirationInMinutes` as this in the try block and any exception should be catched by the 153 line, same for other places using the same logic(keep in mind also the comment regarding the util package)

Reply:
Done.